core-foundation/src/number.rs

```diff
@@ -25,6 +25,25 @@ impl_CFTypeDescription!(CFNumber);
 impl_CFComparison!(CFNumber, CFNumberCompare);
 
 impl CFNumber {
+
+    #[inline]
+    pub fn to_i32(&self) -> Option<i32> {
+        unsafe {
+            let mut value: i32 = 0;
+            let ok = CFNumberGetValue(self.0, kCFNumberSInt32Type, mem::transmute(&mut value));
+            if ok { Some(value) } else { None }
+        }
+    }
+
+    #[inline]
+    pub fn to_u32(&self) -> Option<u32> {
+        unsafe {
+            let mut value: u32 = 0;
+            let ok = CFNumberGetValue(self.0, kCFNumberIntType, mem::transmute(&mut value));
```

Comment:
I think you want kCFNumberSInt32Type and cast the resulting i32 to u32. Otherwise I'm worried that the resulting number might be different sizes on 32 vs. 64 bit architectures.